examples/measure.py

```diff
@@ -0,0 +1,129 @@
+i_index = getIndex("index", cat)
+i_xmax = getIndex("xmax", cat)
+i_ymax = getIndex("ymax", cat)
+i_xmin = getIndex("xmin", cat)
+i_ymin = getIndex("ymin", cat)
+
+schema = afwTable.SourceTable.makeMinimalSchema()
+schema.addField("centroid_x", type=float)
+schema.addField("centroid_y", type=float)
+control = shape.HsmShapeBjControl()
+alg = shape.HsmShapeBjAlgorithm
+#control = measBase.SdssShapeControl()
+#alg = measBase.SdssShapeAlgorithm
+plugin = alg(control, "", schema)
+success = 0
+failed = 0
+count = 0
+startTime = time.time()
+for item in cat.data:
+    bbox = afwGeom.Box2I(afwGeom.Point2I(item[i_xmin], item[i_ymin]), afwGeom.Point2I(item[i_xmax], item[i_ymax]))
+    subexp = afwImage.ExposureF(exp, bbox)
+    #subarray = subexp.getMaskedImage().getImage().getArray()
+    #subarray = numpy.repeat(numpy.repeat(subarray,2, axis=0), 2, axis=1) 
+    #subexp = afwImage.ExposureF(afwImage.MaskedImageF(afwImage.ImageF(subarray)))
+    if psf_dict:
+        psf_number = item[i_psf_number]
+        i = psf_dict[psf_number]
+        data = psf_hdus[i].data.astype(numpy.float64) 
+    else:
+        psf_file = psfFormat % item[i_psf_number]
+        psf_image = afwImage.ImageF(psf_file)
+        data = psf_image.getArray().astype(numpy.float64)
+    kernel = afwMath.FixedKernel(afwImage.ImageD(data))
+    psf = lsst.meas.algorithms.KernelPsf(kernel)
+    subexp.setPsf(psf)
+    subexp.setXY0(afwGeom.Point2I(0,0))
+    detection = SourceDetectionTask()
+    table = afwTable.SourceTable.make(schema)
+    detections = detection.run(table, subexp)
+    detections.sources.defineCentroid("centroid")
+    count = count+1
+    if len(detections.sources) == 0:
+        print "ITEM %d, no sources, "%count, bbox
+        continue
+    record = detections.sources[0]
+    
+    record.setFootprint(detections.fpSets.positive.getFootprints()[0])
+    cen = record.getFootprint().getPeaks()[0].getCentroid()
+    record.set('centroid_x', cen.getX())
+    record.set('centroid_y', cen.getY())
+    try:
+        plugin.measure(record, subexp)
+        print item[i_index], record.get("_e1"), record.get("_e2")
+        success = success+1
+    except lsst.pex.exceptions.wrappers.RuntimeError as e:
+        if e.message.find("PSF") > 0:
+            subexp.writeFits("psf_exp_%d.fits"%item[i_index])
+            subexp.getPsf().computeImage(cen).writeFits("psf_psf_%d.fits"%item[i_index])
+        if e.message.find("iteration") > 0:
+            subexp.writeFits("iter_exp_%d.fits"%item[i_index])
+            subexp.getPsf().computeImage(cen).writeFits("iter_psf_%d.fits"%item[i_index])
+        if e.message.find("all 0") > 0:
+            import pdb
+            pdb.set_trace()
+            subexp.writeFits("zero_exp_%d.fits"%item[i_index])
+            subexp.getPsf().computeImage(cen).writeFits("zero_psf_%d.fits"%item[i_index])
+ 
+        print item[i_index], e.message
+        failed = failed + 1
+    except BaseException as e:
```

Comment:
You probably want to catch something more specific than `BaseException` here - even `Exception` would be a better choice (note that `BaseException` will also catch e.g. `KeyboardInterrupt`, which would make the script impossible to kill).